beacon.go

```diff
@@ -17,110 +17,164 @@
 package beacon
 
 import (
+	"context"
 	"encoding/xml"
 	"errors"
-	"golang.org/x/oauth2/google"
-	"google.golang.org/api/genomics/v1"
-	"google.golang.org/appengine"
+	"fmt"
 	"net/http"
+	"os"
 	"strconv"
+
+	"cloud.google.com/go/bigquery"
+	"google.golang.org/appengine"
 )
 
 type beaconConfig struct {
-	variantSetIds []string
+	projectID string
+	tableId   string
 }
 
+const (
+	projectKey = "GOOGLE_CLOUD_PROJECT"
+	bqTableKey = "GOOGLE_BIGQUERY_TABLE"
+)
+
 var config = beaconConfig{
-	variantSetIds: []string{"3049512673186936334"},
+	projectID: os.Getenv(projectKey),
+	tableId:   os.Getenv(bqTableKey),
 }
 
 func init() {
 	http.HandleFunc("/", handler)
 }
 
-func requestToSearch(r *http.Request) (*genomics.SearchVariantsRequest, string, error) {
-	allele := r.FormValue("allele")
-	search := &genomics.SearchVariantsRequest{
-		ReferenceName: r.FormValue("chromosome"),
-		VariantSetIds: config.variantSetIds,
+func handler(w http.ResponseWriter, r *http.Request) {
+	if err := validateServerConfig(); err != nil {
+		writeError(w, fmt.Errorf("validating server config: %v", err))
```

Comment:
s/config/configuration

Errors are for users and should be free of slang or shortforms

Reply:
done